src/main/resources/static/assets/js/sdh.js

```diff
@@ -0,0 +1,68 @@
+(function() {
+	var app = new Vue({
+		el: '#contents',
+		data: {
+			fields: [{key:'domain', sortable:true}, {key:'date_last_assessed', sortable:true}, {key:'clinician_priority', sortable:true}, {key:'patient_readiness', sortable:true}],
+			filters: {
+			      domain: '',
+			      date_last_assessed: '',
+			      clinician_priority: '',
+			      patient_readiness: ''
+			    },
+	        items: [
+	            { domain: 'Food Security', date_last_assessed: '01/01/2018', clinician_priority: 'High', patient_readiness: 'Not Ready' },
+	            { domain: 'Transportation', date_last_assessed: '01/01/2019', clinician_priority: 'Low', patient_readiness: 'Motivated'},
+	            { domain: 'Housing Instability', date_last_assessed: '10/11/2018', clinician_priority: 'Medium', patient_readiness: 'Ready' },
+	            { domain: 'Utilities', date_last_assessed: '02/01/2019', clinician_priority: 'High', patient_readiness: 'Ready' },
+	            { domain: 'Interpersonal Violence', date_last_assessed: '12/01/2018', clinician_priority: 'Medium', patient_readiness: 'Motivated' },
+	            { domain: 'Education', date_last_assessed: '04/04/2018', clinician_priority: 'Medium', patient_readiness: 'Not Ready' },
+	            { domain: 'Financial Strain', date_last_assessed: '03/17/2012', clinician_priority: 'Low', patient_readiness: 'Ready' }
+	          ]
+		},
+		methods: {
+			_priorityVariant (priority) {
+				if (priority == 'High') {
+					return 'danger';
+				} else if (priority == 'Medium') {
+					return 'warning';
+				} else if (priority == 'Low') {
+					return 'success';
+				}
+		    },			
+			_readinessVariant (readiness) {
+				if (readiness == 'Not Ready') {
+					return 'danger';
+				} else if (readiness == 'Motivated') {
+					return 'warning';
+				} else if (readiness == 'Ready') {
+					return 'success';
+				}
+		    }			
+		},
+		computed: {
+		    filtered () {
+		        const filtered = this.items.filter(item => {
+		          return Object.keys(this.filters).every(key =>
+		              String(item[key].toLowerCase()).includes(this.filters[key].toLowerCase()));
+		        });
+		        return filtered.length > 0 ? filtered : [{
+		          domain: '',
+			      date_last_assessed: '',
+		          clinician_priority: '',
+		          patient_readiness: ''
+		        }];
+		    },
+		    emphasized () {
+		    	const { filtered, _priorityVariant, _readinessVariant } = this;
+		    	const emphasized = this.filtered.map(item => {
+		    		let tmp=item;
+		    		let clinicianPriorityEmphasis = _priorityVariant(item.clinician_priority);
+		    		let patientReadinessEmphasis = _readinessVariant(item.patient_readiness);
+		    		tmp._cellVariants = {'clinician_priority': clinicianPriorityEmphasis, 'patient_readiness': patientReadinessEmphasis};
+		    		return tmp;
```

Comment:
Note that your `map` callback is going to alter the original objects, because `item` is assigned to `tmp` by reference. If you really want a copy and the objects are shallow, you can use `Object.assign`:

```javascript
const emphasized = filtered.map(item => {
  let tmp = Object.assign({}, item);
  tmp._cellVariants = {
    clinician_priority: _priorityVariant(item.clinician_priority),
    patient_readiness: _readinessVariant(item.patient_readiness)
  };
  return tmp;
});
```
If the code were being transpiled, you could do something like this:

```javascript
const emphasized = filtered.map(item => {
  let tmp = { ...item }; // copies properties using object spread
  //...
}); 
```

Reply:
By transpiled, you mean compatible with ES6 with the second example using that syntax? I'm not sure how to set up a transpiler or if one already is. Is this something I should look into?

Reply:
By transpiled, I mean running the code through Webpack and Babel to convert it to something that most browsers can understand. Object rest/spread is an ES2018 feature that isn't widely supported by browsers yet.

As simple as this is currently, it's probably not worth setting up Webpack to get some syntactic sugar. If things get more complicated, it may be worth revisiting the issue.